Source/JavaScriptCore/builtins/StringPrototype.js

```diff
@@ -234,6 +234,10 @@ function hasObservableSideEffectsForStringReplace(regexp, replacer)
     if (regexpUnicode !== @regExpProtoUnicodeGetter)
         return true;
 
+    var regexpUnicode = @tryGetById(regexp, "unicodeSets");
+    if (regexpUnicode !== @regExpProtoUnicodeGetter)
+        return true;
+
```

Comment:
You need to add optimization for `unicodeSets` via watchpoint. See JSGlobalObject.cpp etc. This is comparing with `@regExpProtoUnicodeGetter`, which is `unicode` getter.

Reply:
Also need to add optimization done for `unicode` getter etc. Like, DFGAbstractInterpreterInlines.h etc. for example.

Reply:
See `DFGFixupPhase.cpp` and `DFGAbstractInterpreterInlines.h` too. Let's grep with `propertyNames->global` for example.

Reply:
I fixed these.